added panda_recorder.py with class for adding a recorder to games

FPVAirsoft opened the pull request.

rayanalysis commented:
Hi there, thanks for this new panda3d-screenrecorder PR! It's cool to see some new functionality being made available here.

I noticed that the PR does not have a summary -- did you happen to run any tests on this implementation? Resolution changing tests, different OS or platform tests, that sort of thing?

I noticed that the instantiation routine has also changed. That will require rewriting the Usage page and updating people on the new usage conventions.

I'm excited by the RAM mode functionality but I don't think I can accept this PR as-is. If you're still interested, let me know your thoughts on my initial review here.

FPVAirsoft commented:
Hi!

I fixed the ram-mode issue, removed pillow, and went a step further and made it able to record in real-time (so that the video encodes and saves to disk as you're playing, instead of after) using a threaded encode. 

I tested it on my setup only (windows) and don't have access to test on linux.  Different resolutions were tested, there is setup logic for getting the resolution of the buffer and setting video resolution.

This is only my second PR ever so please forgive my bad manners. 

The new usage I think is more flexible for using buffers and program structures like in my game, but there's also the option to pull the key updates into your original (which it looks like you've made some updates to). 

Since I wasn't sure if you'd want to actually move forward with my structure or just update yours, I wanted to show you the functionality and then we could decide how to proceed.
